libpldmresponder/fru.cpp

```diff
@@ -77,32 +85,29 @@ void FruImpl::updateAssociationTree(const dbus::ObjectValueTree& objects,
         {
             if (objToEntityNode.contains(currPath))
             {
-                pldm_entity node =
-                    pldm_entity_extract(objToEntityNode.at(currPath));
+                pldm_entity_node* node = nullptr;
+                pldm_find_entity_ref_in_tree(
+                    entityTree, objToEntityNode.at(currPath), &node);
+                pldm_entity node_entity = pldm_entity_extract(node);
                 if (pldm_entity_association_tree_find_with_locality(
-                        entityTree, &node, false))
+                        entityTree, &node_entity, false))
                 {
                     break;
                 }
             }
             else
             {
-                if (!objects.contains(currPath))
-                {
-                    break;
-                }
-
-                auto entityPtr = getEntityByObjectPath(objects.at(currPath));
-                if (!entityPtr)
+                pldm_entity entity = getEntityByObjectPath(objects, currPath);
```

Comment:
if **getEntityByObjectPath** is returning empty object then this 3 condition you are comparing instead why can't we use earlier mechanism std::optional<pldm_entity> as return type then you are comparing only one condition with  null_ptr